job-dsl-plugin/src/main/groovy/javaposse/jobdsl/plugin/actions/GeneratedObjectsRunAction.java

```diff
@@ -46,4 +46,9 @@ public void onAttached(Run run) {
     public Collection<T> getModifiedObjects() {
         return modifiedObjects == null ? null : new TreeSet<>(modifiedObjects);
     }
+
+    public void addModifiedObjects(Collection<T> modifiedObjects) {
+        this.modifiedObjects.addAll(modifiedObjects);
```

Comment:
Implementation of `getModifiedObjects` suggests `this.modifiedObjects` might be `null` in certain conditions. I cannot tell it is guaranteed not to cause a problem here so better handle `null` or fail explicitly the method is not supposed to be called on deserialized object.

Reply:
It is a good point. I use it only when Action already exists during build - so it was created by previous job-dsl step in current build. But it is true that public method can be misused or it can cause problems in unexpected situations (for example pipeline build resume after jenkins crush/restart). I believe it will not hurt to put a check.